jdk/src/share/classes/sun/security/ssl/CipherSuite.java

```diff
@@ -874,6 +870,39 @@ enum CipherSuite {
 
     final boolean exportable;
 
+    private static final Map<Integer, CipherSuite> cipherSuiteIds;
+    private static final Map<String, CipherSuite> cipherSuiteNames;
+    private static final List<CipherSuite> allowedCipherSuites;
+    private static final List<CipherSuite> defaultCipherSuites;
+
+    static {
+        Map<Integer, CipherSuite> ids = new HashMap<>();
+        Map<String, CipherSuite> names = new HashMap<>();
+        List<CipherSuite> allowedCS = new ArrayList<>();
+        List<CipherSuite> defaultCS = new ArrayList<>();
+
+        for(CipherSuite cs : CipherSuite.values()) {
+            ids.put(cs.id, cs);
+            names.put(cs.name, cs);
+            for (String alias : cs.aliases) {
+                names.put(alias, cs);
+            }
+
+            if (!cs.supportedProtocols.isEmpty()) {
+                allowedCS.add(cs);
+            }
+
+            if (cs.isDefaultEnabled) {
+                defaultCS.add(cs);
+            }
+        }
+
+        cipherSuiteIds = Collections.unmodifiableMap(ids);
+        cipherSuiteNames = Collections.unmodifiableMap(names);
+        allowedCipherSuites = Collections.unmodifiableList(allowedCS);
+        defaultCipherSuites = Collections.unmodifiableList(defaultCS);
+    }
```

Comment:
Agree that in this case `Collections.unmodifiableMap()`/`Collections.unmodifiableList()` are a good replacement for the missing `copyOf()`. And it should also be faster, because it avoids a copy :)